### PR TITLE
Update hapi to latest version

### PR DESCRIPTION
**Pull request by rfk**
for CI at the moment

### CODE REVIEW

**Comment by rfk** [comment]
OK, the tests seem to be going green, calling this ready for r?

File: lib/routes/devices-sessions.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ module.exports = (log, db, config, customs, push, devices) => {
 
         // Some additional, slightly tricky validation to detect bad public keys.
         if (payload.pushPublicKey && ! push.isValidPublicKey(payload.pushPublicKey)) {
-          throw error.invalidRequestParameter('invalid pushPublicKey')
+          return reply(error.invalidRequestParameter('invalid pushPublicKey'))
```
Comment (rfk) [inline lib/routes/devices-sessions.js]: This pattern of throwing a boom-compatible Error in the top level of the handler function no longer seems to work.  The docs seem to recommend either directly replying with the error, return returning a promise that will reject with the error.  This seems like the smaller change.

File: test/local/server.js [hunk 2]
```diff
@@ -128,7 +128,6 @@ describe('lib/server', () => {
           assert.equal(args[0], log.begin.args[0][1])
           assert.ok(args[1])
           assert.equal(args[1].isBoom, undefined)
-          assert.equal(args[1].message, undefined)
```
Comment (rfk) [inline test/local/server.js]: `Response` objects now have a `.message()` method that can be used to set the HTTP status-line message

**Comment by rfk** [comment]
(btw this is targeting train-95 branch, for point release and then merge to master)

File: npm-shrinkwrap.json [hunk 3]
```diff
@@ -328,9 +328,9 @@
       "resolved": "git+https://github.com/mozilla/eslint-plugin-fxa.git#41504c9dd30e8b52900c15b524946aa0428aef95"
     },
     "fxa-auth-db-mysql": {
-      "version": "1.95.0",
+      "version": "1.95.1",
```
Comment (philbooth) [inline npm-shrinkwrap.json]: @vladikoff made a `1.95.2` tag directly from master yesterday so it looks like you might need to rebase the `train-95` branch to that before opening this PR?

```
* |  (tag: v1.95.2) 5df01c70 vladikoff Release v1.95.2
* |  2e8e6747 Vlad Filippov feat(db): allow BMP chars in device name (#2053) r=rfk,jbuck
| *    57a7f996 Phil Booth Merge branch 'master' into phil/2106
| |\
| |/
|/|
* |    c6486ba0 Ryan Kelly Merge branch 'train-95'
|\ \
| * |  (HEAD -> train-95, tag: v1.95.1, origin/train-95) 187943d8 Ryan Kelly Release v1.95.1
```

**Pull request merged**